src/ServiceControl/Infrastructure/RavenDB/Expiration/ExpiredDocumentsCleanerBundle.cs

```diff
@@ -50,10 +50,14 @@ public void Dispose()
         {
             if (timer != null)
             {
-                using (var manualResetEvent = new ManualResetEvent(false))
+                lock (timer)
```

Comment:
Wat dis?

Reply:
This change is necessary because for some reasons Raven 3.5 calls dispose twice (not sure if sequentially or if it can happen in parallel). If you call

```
timer.Dispose(manualResetEvent);
```

twice it will lock forever. This lock protects the `timer = null` statement to ensure that the underlying timer dispose is attempted only once.

Reply:
:+1: